Remove Stackdriver, Elasticsearch content from Tasks/Monitoring.. and Concepts/Cluster Admin/Logging Arch (#16203)
kubernetes/website: content/en/docs/concepts/cluster-administration/events.md

---
reviewers:
- piosz
- x13n
content_template: templates/concept
title: Events in Kubernetes
---

{{% capture overview %}}

> Kubernetes events are objects that provide insight into what is happening

Reviewer: Consider using a glossary definition shortcode here, that automatically includes either the short or full version of the glossary text for this object.

Author: done

inside a cluster, such as what decisions were made by scheduler or why some
pods were evicted from the node. 
{{% /capture %}}

{{% capture body %}}

> Since events are API objects, they are stored in the {{< glossary_tooltip text="apiserver" term_id="kube-apiserver" >}} on master. To

Reviewer: ```suggestion
Since Events are API objects, they are stored in the {{< glossary_tooltip text="apiserver" term_id="kube-apiserver" >}} on the control plane. To
```

Author: done

> avoid filling up master's disk, a retention policy is enforced: events are

Reviewer: ```suggestion
avoid filling up storage, a retention policy is enforced: events are
```
(there might be more than one server in the control plane, and it might not use its local disk for persistence)

Author: done - thanks! Need to **really** brush up on my Kubernetes knowledge!

removed one hour after the last occurrence. To provide longer history

> and aggregation capabilities, a third-party solution should be installed

Reviewer: ```suggestion
and aggregation capabilities, you can install a third-party solution
```

Author: Aaargghh foiled again by the Style Guide!

to capture events. Exploring third-party solutions is beyond the scope of the
Kubernetes documentation. You may wish to look at the [CNCF Cloud Native Interactive Landscape](https://landscape.cncf.io)
for projects that can consume and aggregate Kubernetes events.

{{% /capture %}}

{{% capture whatsnext %}}

> You can read more about using events for debugging your application in the [Application Introspection and Debugging](/docs/tasks/debug-application-cluster/debug-application-introspection/) section.

Reviewer: ```suggestion
* Read about using Events for debugging your application in the [Application Introspection and Debugging](/docs/tasks/debug-application-cluster/debug-application-introspection/) section.
```

Author: done
